convert2rhel/hostmetering.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright(C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Integrates host-metering with RHEL.
+https://github.com/RedHatInsights/host-metering/
+"""
+
+__metaclass__ = type
+
+import logging
+import os
+
+from convert2rhel import systeminfo
+from convert2rhel.pkghandler import call_yum_cmd
+from convert2rhel.subscription import get_rhsm_facts
+from convert2rhel.systeminfo import system_info
+from convert2rhel.utils import run_subprocess
+
+
+logger = logging.getLogger(__name__)
+
+
+def is_running_on_hyperscaler(rhsm_facts):
+    """
+    Check if the system is running on hyperscaler. Currently supported
+    hyperscalers are aws, azure and gcp.
+
+    :param rhsm_facts: Facts about the system from RHSM.
+    :type rhsm_facts: dict
+    :return: True if the system is running on hyperscaler, False otherwise.
+    :rtype: bool
+    """
+    is_aws = rhsm_facts.get("aws_instance_id")
+    is_azure = rhsm_facts.get("azure_instance_id")
+    is_gcp = rhsm_facts.get("gcp_instance_id")
+    return any([is_aws, is_azure, is_gcp])
+
+
+def configure_host_metering():
+    """
+    Decide whether to install, enable and start host-metering on the system based on the
+    CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable.
+
+    Behavior can be controlled CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable:
+    - "auto": host-metering will be configured based on the above conditions
+    - empty: host-metering will not be configured
```

Comment:
Probably we could remove this from here, or reword it to say that if the env var is not present, nothing will happen. Having the keyword `empty` here caused me a confusion to think that we could potentially say `CONVERT2RHEL_CONFIGURE_HOST_METERING=empty`.